woocommerce/woocommerce-ios
# Orders Details: Use common class for Country and State selector in Order Details

Part of https://github.com/woocommerce/woocommerce-ios/issues/6667

### Description
`CountrySelectorCommand` and `StateSelectorCommand` share the majority of its code. To avoid the downsides of having duplicated code, we're abstracting these into one common class.

### Changes:
- Create a common `AddressSelectorCommandProtocol` that `Country` and `StateOfACountry` conforms to.
- Collapse `CountrySelectorCommand` and `StateSelectorCommand` into a common `AddressSelectorCommand` class. Its implementation details use the common `AddressSelectorCommandProtocol` rather than the specifics of each model.
- Now that `CountrySelectorViewModel` and `StateSelectorViewModel` use `AddressSelectorCommand`, these could be collapsed as well, but will be part of a different PR.

### Testing instructions
We have not changed the functionality, but the following flow can be used to test the changes where these selectors are used:

1. Go to `Orders` > `+` > `Create Order` > Fill the minimum necessary fields
2. Tap on `Add Customer Details` > Fill the minimum necessary fields
3. Confirm that the `Country` and `State` fields works as expected by saving new or editing existing data, for example:
- Select US > Select a predefined `State` from the list. Save changes.
- Select a non-US country, write down something in `State`, or leave it empty. Save Changes.
- Tap on `Add a different shipping address` and fill out `Country` and `State` again with different values. Save changes.
- When tapping `Country`, confirm that the search bar at the top works by filtering countries.
- When selecting US as `Country` and tapping `State`, confirm that the search bar at the top works by filtering states.
4. Confirm how the Order reflects all changes upon saving.
